### PR TITLE
type-registry: add DateString type

### PR DESCRIPTION
**Pull request by kjdelisle**
#### Related issues
https://github.com/strongloop/loopback-datasource-juggler/pull/1356
https://github.com/strongloop/loopback-connector-mysql/issues/149

### Checklist

- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style
  guide](http://loopback.io/doc/en/contrib/style-guide.html)

### CODE REVIEW

**Comment by kjdelisle** [comment]
@slnode test please

**Comment by kjdelisle** [comment]
CI failure is due to a dependency resolution failure on the server, not sure why we keep getting these random hiccups....

**Comment by kjdelisle** [comment]
@slnode test please

File: lib/specgen/type-registry.js [hunk 1]
```diff
@@ -25,6 +25,7 @@ function TypeRegistry() {
       lng: {type: 'number'},
     },
   });
+  this.registerLoopbackType('DateString', {type: 'string', format: 'date-time'});
```
Comment (bajtos) [inline lib/specgen/type-registry.js]: I find this suspicious. The name `DateString` suggests the value should contain only a date with no time information. In that case, the swagger type should use format `date`. See [Swagger Data Types](https://github.com/OAI/OpenAPI-Specification/blob/ae9322eb2df1555acf3163e30cd84779d98afec5/versions/2.0.md#data-types).

File: test/specgen/schema-builder.test.js [hunk 2]
```diff
@@ -135,6 +135,12 @@ describe('schema-builder', function() {
       out: {$ref: '#/definitions/GeoPoint', description: '{lng: 10, lat: 20}'}},
     {in: {type: 'GEOPOINT', doc: '{lng: 10, lat: 20}'},
       out: {$ref: '#/definitions/GeoPoint', description: '{lng: 10, lat: 20}'}},
+    {in: {type: 'DateString'},
+      out: {$ref: '#/definitions/DateString'}},
```
Comment (bajtos) [inline test/specgen/schema-builder.test.js]: This looks wrong to me. Where is `#/definitions/DateString` defined?  Why are we referring to a definition, when `DateString` is defined as a plain formatted string?

I would expect a test case similar to ` describeTestCases('for string types', [` above:

```js
     {in: 'number', out: {type: 'number', format: 'double'}},
     {in: 'date', out: {type: 'string', format: 'date-time'}},
     // add this
     {in: 'DateString', out: {type: 'string', format: 'date'}},
```

File: lib/specgen/type-registry.js (hunk 1 above)
Comment (kjdelisle) [inline lib/specgen/type-registry.js]: This type is meant to be the string representation of a Date object (literally, Date-String), without automatic application of timezones. It _can_ be a date exclusively, but it supports date-time values.

File: test/specgen/schema-builder.test.js (hunk 2 above)
Comment (kjdelisle) [inline test/specgen/schema-builder.test.js]: The mapping of the type is generated as a result of the registration of the LoopBack type, just as GeoPoint is. The definition itself will be available once https://github.com/strongloop/loopback-datasource-juggler/pull/1356 lands.

File: lib/specgen/type-registry.js (hunk 1 above)
Comment (bajtos) [inline lib/specgen/type-registry.js]: Yeah, I figured this out myself too. As I commented in juggler PR, I would prefer `DateTimeString`, but let's move on.

File: test/specgen/schema-builder.test.js (hunk 2 above)
Comment (bajtos) [inline test/specgen/schema-builder.test.js]: Well, GeoPoint is an object-based type with multiple properties, that's why it's registered in `#/definitions` from what I remember. `DateString` is the same thing as `Date` when it comes to swagger representation, so why are we emitting different swagger for it?

Here is a corrected code from my comment above:

```js
     {in: 'number', out: {type: 'number', format: 'double'}},
     {in: 'date', out: {type: 'string', format: 'date-time'}},
     // add this
     {in: 'DateString', out: {type: 'string', format: 'date-time'}},
```

Anyways, I see that `ObjectID` is registered via `registerLoopbackType ` too ([source](https://github.com/strongloop/loopback-swagger/blob/feb97debb56134edbb22c7f6761583d5f1f93584/lib/specgen/type-registry.js#L21)), even though it's just a string with a pattern matcher, which makes my call for consistency rather weak I guess.

**Pull request merged**